homeassistant/components/sunricher_dali/scene.py

```diff
@@ -0,0 +1,116 @@
+"""Support for DALI Center Scene entities."""
+
+import logging
+from typing import Any
+
+from PySrDaliGateway import CallbackEventType, Scene
+from PySrDaliGateway.helper import gen_device_unique_id, gen_group_unique_id
+
+from homeassistant.components.scene import Scene as SceneEntity
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import entity_registry as er
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .const import DOMAIN
+from .types import DaliCenterConfigEntry
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 1
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: DaliCenterConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up DALI Center scene entities from config entry."""
+    scenes = entry.runtime_data.scenes
+
+    scene_entities: list[DaliCenterScene] = []
+    for scene in scenes:
+        try:
+            scene_details = await scene.read_scene()
+            scene_entities.append(DaliCenterScene(scene, scene_details))
+        except (OSError, ValueError, KeyError):
+            _LOGGER.exception(
+                "Failed to read scene details for %s, skipping scene",
+                scene.name,
+            )
+
+    if scene_entities:
+        async_add_entities(scene_entities)
+
+
+class DaliCenterScene(SceneEntity):
+    """Representation of a DALI Center Scene."""
+
+    _attr_has_entity_name = True
+    _attr_available = True
+
+    def __init__(self, scene: Scene, scene_details: dict[str, Any]) -> None:
+        """Initialize the DALI scene."""
+
+        self._scene = scene
+        self._attr_name = scene.name
+        self._attr_unique_id = scene.unique_id
+        self._scene_details = scene_details
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, scene.gw_sn)},
+        )
+        self._attr_extra_state_attributes = {
+            "gateway_sn": scene.gw_sn,
+            "scene_id": scene.scene_id,
+            "area_id": scene_details["area_id"],
+            "channel": scene_details["channel"],
+            "entity_id": [],
+        }
```

Comment:
What does the user need this for?
serial number is already in the device
scene id?
area_id would also be in the device?
What's channel for?